libraries/node-core-library/src/Objects.ts

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as Objects from './objects/index';
```

Comment:
Not opposed to this new sub-folder structure since it allows for breaking out functions into individual files. But what is the benefit of the index file vs just including them here?

Reply:
This is how I make the public API contract be `Objects.areDeepEqual(...)` instead of `objectsAreDeepEqual(...)`, while still letting the tree-shaker do its thing. Technically since the index file is only reexporting from one file it is currently redundant, but if we add more exported functions in separate files that will necessitate the existence of the index, so this is future-proofing.